internal/controllers/cluster/cluster_controller.go

```diff
@@ -141,6 +141,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 		return ctrl.Result{}, err
 	}
 
+	s := &scope{
+		cluster: cluster,
+	}
+
 	// Initialize the patch helper.
 	patchHelper, err := patch.NewHelper(cluster, r.Client)
 	if err != nil {
```

Comment:
We should probably also set retRes to ctrl.Result at the end of defer if we have a reterr